src/wallet/ser.rs

```diff
@@ -117,8 +141,8 @@ impl<'a> Writeable for ChangeSetSerWrapper<'a, BTreeSet<(ConfirmationBlockTime,
 		let len = BigSize(self.0.len() as u64);
 		len.write(writer)?;
 		for (time, txid) in self.0.iter() {
-			ChangeSetSerWrapper(time).write(writer)?;
-			txid.write(writer)?;
+			write_len_prefixed_field!(writer, ChangeSetSerWrapper(time))?;
+			write_len_prefixed_field!(writer, txid)?;
 		}
```

Comment:
Hmm... right, on second thought we probably don't need to use `Iterable` or `WithoutLength`. But shouldn't we not write the length prefix for the `BTreeSet` size? Any `BTreeSet` or `BTreeMap` will always be written in the context of an `encode_tlv_stream`, so we'd end up writing the length twice. In LDK, we use `WithoutLength` to avoid this, but that's only because the serialization format predated the use of `encode_tlv_stream`.